src/unlimited.cpp

```diff
@@ -1148,12 +1147,24 @@ void IsChainNearlySyncdInit()
         fIsChainNearlySyncd = true;
       }
 }
+
 bool IsChainNearlySyncd()
 {
     LOCK(cs_ischainnearlysyncd);
     return fIsChainNearlySyncd;
 }
 
+uint64_t LargestBlockSeen(uint64_t nBlockSize)
+{
+     // eturn the largest block size that we have seen since startup
+     uint64_t nSize = nLargestBlockSeen.load();
+     if (nBlockSize > nSize) {
+         nLargestBlockSeen.store(nBlockSize);
```

Comment:
Guys, seriously, this is invalid. I need to use CAS and that was mentioned already.